src/Numerics/DGMethods/DGFVModel_kernels.jl

```diff
@@ -0,0 +1,800 @@
+        balance_law::BalanceLaw,
+        ::Val{info},
+        ::Val{nvertelem},
+        ::Val{periodicstack},
+        ::VerticalDirection,
+        numerical_flux_first_order,
+        tendency,
+        state_prognostic,
+        state_auxiliary,
+        vgeo,
+        sgeo,
+        t,
+        vmap⁻,
+        vmap⁺,
+        elemtobndy,
+        elems,
+        α,
+    )
+
+Compute kernel for evaluating the interface tendencies using vertical FVM
+reconstructions with a DG method of the form:
+
+∫ₑ ψ⋅ ∂q/∂t dx - ∫ₑ ∇ψ⋅(Fⁱⁿᵛ + Fᵛⁱˢᶜ) dx + ∮ₑ n̂ ψ⋅(Fⁱⁿᵛ⋆ + Fᵛⁱˢᶜ⋆) dS,
+
+or equivalently in matrix form:
+
+dQ/dt = M⁻¹(MS + DᵀM(Fⁱⁿᵛ + Fᵛⁱˢᶜ) + ∑ᶠ LᵀMf(Fⁱⁿᵛ⋆ + Fᵛⁱˢᶜ⋆)).
+
+This kernel computes the surface terms: M⁻¹ ∑ᶠ LᵀMf(Fⁱⁿᵛ⋆ + Fᵛⁱˢᶜ⋆)), where M
+is the mass matrix, Mf is the face mass matrix, L is an interpolator from
+volume to face, and Fⁱⁿᵛ⋆, Fᵛⁱˢᶜ⋆ are the numerical fluxes for the inviscid
+and viscous fluxes, respectively.
+
+A finite volume reconstruction is used to construction `Fⁱⁿᵛ⋆`
+""" vert_fvm_interface_tendency!
+@kernel function vert_fvm_interface_tendency!(
+    balance_law::BalanceLaw,
+    ::Val{info},
+    ::Val{nvertelem},
+    ::Val{periodicstack},
+    ::VerticalDirection,
+    reconstruction!,
+    numerical_flux_first_order,
+    numerical_flux_second_order,
+    tendency,
+    state_prognostic,
+    state_gradient_flux,
+    state_auxiliary,
+    vgeo,
+    sgeo,
+    t,
+    elemtobndy,
+    elems,
+    α,
+) where {info, nvertelem, periodicstack}
+    @uniform begin
+        dim = info.dim
+        FT = eltype(state_prognostic)
+        num_state_prognostic = number_states(balance_law, Prognostic())
+        num_state_primitive = number_states(balance_law, Primitive())
+        num_state_auxiliary = number_states(balance_law, Auxiliary())
+        num_state_gradient_flux = number_states(balance_law, GradientFlux())
+        num_state_hyperdiffusive = number_states(balance_law, Hyperdiffusive())
+        @assert num_state_hyperdiffusive == 0
+
+        nface = info.nface
+        Np = info.Np
+        Nqk = info.Nqk # can only be 1 for the FVM method!
+        @assert Nqk == 1
+
+        # We only have the vertical faces
+        faces = (nface - 1):nface
+
+        stencil_width = width(reconstruction!)
+
+        # If this fails the `@nif` below needs to change
+        @assert stencil_width < 4
+
+        # In the case of stencil_width = 0 we still need two values to evaluate
+        # the fluxes, so the minimum stencil diameter is 2
+        stencil_diameter = max(2, 2stencil_width + 1)
+
+        # Value in the stencil that corresponds to the top face with respect to
+        # face being updated
+        stencil_center = max(stencil_width, 1) + 1
+
+        # 1 → cell i, face i - 1/2
+        # 2 → cell i, face i + 1/2
+        local_state_face_prognostic = ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_prognostic}, FT}(undef)
+        end
```

Comment:
Is it an array of stencil_diameter MArray{Tuple{num_state_prognostic}, FT}(undef) ? stencil_diameter=2 for FVLinear, do we need 5 for WENO?
Do we need that many face prognostic?
What are these 1, 2 in the comments?

Reply:
For WENO5 you will set the width to `2` in the `FVReconstructions` module for a  new `FVWENO` type. The diameter will be automatically computed.

We can probably shrink the `local_state_prognostic` to `width + 1` later. @mwarusz told me you still need the full auxiliary though. Once we have a fully working code, we will need to do some optimization passes, but I was going for correctness and not terrible performance to start. (I've included some of these in the comment section entitled `Optimization ideas`.)

The `1` and `2` refer to the bottom and top faces of the element being constructed (this is what I tried to indicate in the comment). I've updated the comment let me know if you have other suggestions.

Reply:
I see that you were asking about face prognostic, yes you are write. This was a typo on my part. Good catch! Will change to `2`